7-writing-plans/README.md

```diff
@@ -4,162 +4,160 @@
-[
-  {
-    "node": "node1",
-    "status": "success",
-    "result": {
-      "stdout": " 23:35:48 up 2 min,  0 users,  load average: 0.10, 0.09, 0.04\n",
-      "stderr": "",
-      "exit_code": 0
+Create a simple plan that runs a command on a list of nodes.
+
+1. Save the following as `modules/exercise7/plans/command.pp`:
+
+    ```puppet
+    plan exercise7::command (TargetSpec $nodes) {
+      run_command("uptime", $nodes)
     }
-  }
-]
-```
+    ```
 
-Note that:
+2. Run the plan:
 
-* `nodes` is passed as an argument like any other, rather than using a flag. This makes plans flexible when it comes to taking lists of different types of nodes for instance, or generating the list of nodes in code within the plan.
+    ```
+    $ bolt plan run exercise7::command nodes=all --modulepath ./modules
+    2018-02-16T15:35:47.843668 INFO   Bolt::Executor: Starting command run 'uptime' on ["node1"]
+    2018-02-16T15:35:48.154690 INFO   Bolt::Executor: Ran command 'uptime' on 1 node with 0 failures
+    [
+      {
+        "node": "node1",
+        "status": "success",
+        "result": {
+          "stdout": " 23:35:48 up 2 min,  0 users,  load average: 0.10, 0.09, 0.04\n",
+          "stderr": "",
+          "exit_code": 0
+        }
+      }
+    ]
+    ```
 
-    As a convention, we recommend using the `TargetSpec` type to denote nodes; it allows passing a single string describing a target URI or a comma-separated list of strings as supported by the `--nodes` argument to other commands. It also accepts an array of Targets, as resolved by calling the [`get_targets` method](https://puppet.com/docs/bolt/0.x/writing_plans.html#calling-basic-plan-functions), allowing you to iterate over Targets without needing to do your own string splitting, or as resolved from a group in an [inventory file](https://puppet.com/docs/bolt/0.x/inventory_file.html).
+    **Note:**
+
+    * `nodes` is passed as an argument like any other, rather than a flag. This makes plans flexible when it comes to taking lists of different types of nodes or generating the list of nodes in code within the plan.
+
+    * Use the `TargetSpec` type to denote nodes; it allows passing a single string describing a target URI or a comma-separated list of strings as supported by the `--nodes` argument to other commands. It also accepts an array of Targets, as resolved by calling the [`get_targets` method](https://puppet.com/docs/bolt/0.x/writing_plans.html#calling-basic-plan-functions). You can iterate over Targets without needing to do your own string splitting, or as resolved from a group in an [inventory file](https://puppet.com/docs/bolt/0.x/inventory_file.html).
 
 
 # Write a plan using run_task
+Create a task and then create a plan that uses the task.
+
+1. Save the following task as `modules/exercise7/tasks/write.sh`. The task accepts a filename and some content and saves a file to 1`/tmp`.
```

Comment:
Stray `1` in here

Reply:
This is step 1 of 4.

Reply:
Oh I meant 1`/tmp`, that shouldn't have a 1, should it?

Reply:
I see where you mean, now. Fixed.